src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts

```diff
@@ -138,6 +138,7 @@ export class DatasetService {
   }
 
   private cacheDataStructure(dataType: string, dataStructure: DataStructure) {
+    this.setLastCacheTime(new Date());
```

Comment:
Since this is getting set in the sessionStorage, the Date object gets converted to a string. To consume this, we will need to parse it and what not. To programmatically use date, it is best to stick with the unix timestamp of the date which is number and can be easily parsed and formatted.

```suggestion
    this.setLastCacheTime(Date.now());
```